ui/app/models/ssh/ca-config.js

```diff
@@ -5,16 +5,49 @@
 
 import Model, { attr } from '@ember-data/model';
 import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { withModelValidations } from 'vault/decorators/model-validations';
 
+const validations = {
+  publicKey: [
+    {
+      validator(model) {
+        const { publicKey, privateKey, generateSigningKey } = model;
+        // if generateSigningKey is false, use the second validation to check if both public and private keys are set
+        return (publicKey && privateKey) || (!publicKey && !privateKey) || !generateSigningKey ? true : false;
+      },
+      message: 'Public Key and Private Key are both required if one of them is set.',
```

Comment:
```suggestion
      message: 'ERROR: You must provide a public and private encryption key or leave both unset.',
```
Suggest making the error text more direct.

Reply:
Using suggested text, but not including `ERROR:` as it doesn't follow the current flow of all other validation messages. Again, happy to make a wider codebase change, but should be addressed in another PR/ticket.